### PR TITLE
list column families

### PR DESCRIPTION
**Pull request by jeizsm**
Add list_cf method

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -659,6 +659,33 @@ impl DB {
...
+        unsafe {
+            let ptr = ffi_try!(ffi::rocksdb_list_column_families(opts.inner,
+                                                                 cpath.as_ptr() as *const _,
+                                                                 &mut length));
+
+            vec = Vec::from_raw_parts(ptr, length, length).iter().map(|&ptr| {
+                CString::from_raw(ptr).into_string().unwrap()
+            }).collect();
+        }
+        Ok(vec)
```
Comment (spacejam) [inline src/db.rs]: You can have `Ok(vec)` be the last line in the unsafe block, and the function will return it there. No need to declare `vec` on line 674 before assigning it.

File: tests/test_column_family.rs [hunk 2]
```diff
@@ -62,6 +62,17 @@ pub fn test_column_family() {
             Err(e) => panic!("failed to open db with column family: {}", e),
         }
     }
+
+    // should be able to list a cf
+    {
+        let opts = Options::default();
+        let vec = DB::list_cf(&opts, path);
+        match vec {
+            Ok(vec) => println!("list cf: {:?}", vec),
```
Comment (spacejam) [inline tests/test_column_family.rs]: Could you change the `println!` to an `assert_eq!` that ensures the vec matches what you expect it to be?

**Comment by spacejam** [comment]
Awesome work! Just a couple small changes and this is good for merging :)

**Comment by spacejam** [comment]
Thanks for adding this! I'll cut a new version with these changes now!

**Comment by spacejam** [comment]
`0.8.1` is up on crates.io with these changes :)

**Comment by jeizsm** [comment]
Thank you :)

**Pull request merged**